core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * 
+ */
+package com.netflix.conductor.core.execution.tasks;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.tasks.Task.Status;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.events.ScriptEvaluator;
+import com.netflix.conductor.core.execution.WorkflowExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.script.ScriptException;
+import java.util.*;
+
+/**
+ * @author Manan
+ *
+ */
+public class DoWhile extends WorkflowSystemTask {
+
+	public DoWhile() {
+		super("DO_WHILE");
+	}
+
+	Logger logger = LoggerFactory.getLogger(DoWhile.class);
+
+	@Override
+	public void cancel(Workflow workflow, Task task, WorkflowExecutor executor) {
+		task.setStatus(Status.CANCELED);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean execute(Workflow workflow, Task task, WorkflowExecutor workflowExecutor) {
+		
+		boolean allDone = true;
+		boolean hasFailures = false;
+		StringBuilder failureReason = new StringBuilder();
+		List<WorkflowTask> loopOver = task.getWorkflowTask().getLoopOver();
+		Map<String, Object> output = new HashMap<>();
+		task.getOutputData().put("iteration", task.getIteration());
+
+		for (WorkflowTask workflowTask : loopOver){
+			Task loopOverTask = workflow.getTaskByRefName(workflowTask.getTaskReferenceName());
+			if (loopOverTask == null){
+				//Task is not even scheduled yet
+				allDone = false;
+				break;
+			}
+			Status taskStatus = loopOverTask.getStatus();
+			hasFailures = !taskStatus.isSuccessful();
+			if (hasFailures){
+				failureReason.append(loopOverTask.getReasonForIncompletion()).append(" ");
+			}
+			output.put(workflowTask.getTaskReferenceName(), loopOverTask.getOutputData());
+			allDone = taskStatus.isTerminal();
+			if (!allDone || hasFailures){
+				break;
+			}
+		}
+		task.getOutputData().put(String.valueOf(task.getIteration()), output);
```

Comment:
@shivambansal2222  storing output of each iteration with key as iteration. See if this can be useful for you.

Reply:
Could it not be preferable to store the output directly inside an Array, using the index as information about the iteration?

Reply:
Initially I had thought about it storing array way, but Storing it map way has advantage because it will automatically update the value to latest. Map has inbuilt facility for that. Also output map can be filled partially depend upon which task completed and which task remaining so map is better.

Reply:
Ok, but in this case why do you stringify the iteration? It would not be better to have as key the same type of _iteration_ attribute (integer) ?

Reply:
No specific reason for this. Will change in subsequent commit.